### ISSUE TITLE
feat: Add relations support for collection types

### ISSUE THREAD

**Issue by jaredcwhite** [bridgetownrb/bridgetown#260]
As mentioned in #187, there should be a way to define relations in the config file between collection resources. For example, given a config of:

```yaml
collections:
  actors:
    output: true
    relations:
      has_many: movies
  movies:
    output: true
    relations:
      belongs_to:
        - actors
        - crewmembers
        - studio
  studios:
    output: true
    relations:
      has_many: movies
```

The following data accessors would be available:

`actor.relations.movies`
`movie.relations.actors`
`movie.relations.crewmembers`
`movie.relations.studio`
`studio.relations.movies`

The `belongs_to` type relations are where the relations are explicitly defined, slug-based (although we could make this configurable I guess). `belongs_to` can support solo or multiple relations.

```yaml
# _movies/_1982/blade-runner.md
name: Blade Runner
actors:
  - harrison-ford # _actors/_h/harrison-ford.md
  - rutger-howard # _actors/_r/rutger-howard.md
  - sean-young # _actors/_s/sean-young.md
crewmembers:
  - jordan-cronenweth # _crewmembers/_j/jordan-cronenweth
studio: warner-brothers # _studios/warner-brothers.md
```

I plan to use the Rails/ActiveSupport inflector for singular/plural conversions.

**Comment by jaredcwhite** [comment]
Feedback? Thoughts? Questions?

**Comment by jaredcwhite** [comment]
Resolved in #261
